var/spack/repos/builtin/packages/awscli-v2/package.py

```diff
@@ -6,26 +6,43 @@
 from spack.package import *
 
 
-class AwscliV2(PythonPackage):
+class AwscliV2(Package, PythonPackage):
     """This package provides a unified command line interface to Amazon Web Services."""
 
     homepage = "https://docs.aws.amazon.com/cli"
     url = "https://github.com/aws/aws-cli/archive/refs/tags/2.13.22.tar.gz"
 
     maintainers("climbfuji")
 
+    version("latest", sha256="d045a8baafbe45ab87821b1a1019ba64683dbb2659e1b5ab985f869a52c1bef5")
     version("2.13.22", sha256="dd731a2ba5973f3219f24c8b332a223a29d959493c8a8e93746d65877d02afc1")
 
-    depends_on("python@3.8:", type=("build", "run"))
-    depends_on("py-flit-core@3.7.1:3.8.0", type=("build"))
-    depends_on("py-colorama@0.2.5:0.4.6", type=("build", "run"))
-    depends_on("py-docutils@0.10:0.19", type=("build", "run"))
-    depends_on("py-cryptography@3.3.2:40.0.1", type=("build", "run"))
-    depends_on("py-ruamel-yaml@0.15:0.17.21", type=("build", "run"))
-    depends_on("py-ruamel-yaml-clib@0.2:0.2.7", type=("build", "run"))
-    depends_on("py-prompt-toolkit@3.0.24:3.0.38", type=("build", "run"))
-    depends_on("py-distro@1.5:1.8", type=("build", "run"))
-    depends_on("py-awscrt@0.16.4:0.16.16", type=("build", "run"))
-    depends_on("py-python-dateutil@2.1:2", type=("build", "run"))
-    depends_on("py-jmespath@0.7.1:1.0", type=("build", "run"))
-    depends_on("py-urllib3@1.25.4:1.26", type=("build", "run"))
+    def url_for_version(self, version):
+        if version == Version("latest"):
+            return "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip"
```

Comment:
How do we handle macOS, other Linux architectures, Windows etc?

Reply:
Hm was hoping I could do something similar with MacOS but they're using the .pkg format, which is restrictive in terms of how to install, plus I don't see it being used anywhere else in Spack... Maybe we need to suck it up and install Rust? What do you think?

Reply:
Either that or we make it an external dependency ... somewhat counter our efforts to reduce the number of external dependencies. Let me take another look at the awscli-v2 package, I remember that there were options to configure it with its own internal dependencies or with external packages (from spack etc).

Reply:
From what I can tell, awscli-v2 "really" needs the cryptography module, and py-cryptography "really" needs Rust at build time, _but_ I was able to get py-cryptography to build against rust-bootstrap. I don't know what approach the Spack devs would prefer, since arguably adding a variant to switch from rust to rust-bootstrap is a bit ugly (not that I personally mind)...

Reply:
Do you mind checking with them?

Reply:
I haven't heard anything yet. I forget if I already asked, but what would you think of having awscli-v2 be gcc/clang only as far as our metapackages are concerned (`depends_on("awscli-v2", when="%gcc")` or similar in ewok-env)? I'm not yet clear on whether rust can be built with non-gcc/clang compilers (if that's the case maybe we can fix it in the recipe).

Reply:
That won't work - we need this for the Intel stack, too. I'd rather proceed with rust-boostrap instead of rust, and if spack mainline doesn't like it we can maintain that difference?

Reply:
Hmm... Okay yeah let's do that for now, then if Spack or Rust devs have a better solution, we can circle back later.